jmclient/jmclient/wallet_utils.py

```diff
@@ -97,15 +97,14 @@ def get_wallettool_parser():
                       help='hd wallet path (e.g. m/0/0/0/000)')
     parser.add_option('--key-type',  # note: keep in sync with map_key_type
                       type='choice',
-                      choices=('standard', 'segwit-p2sh'),
+                      choices=('standard', 'segwit-p2sh', 'segwit'),
```

Comment:
I would prefer using the same names as Bitcoin Core uses (for example, in `getnewaddress` RPC) - "legacy", "p2sh-segwit", "bech32". But not sure is it important to keep backwards compatibility here. And "bech32" itself isn't good name, as pay-to-taproot in future will also be a bech32 address. :(

Reply:
"Legacy" also crossed my mind here and is a term I'm familiar with, but I didn't want to mess with existing code/functionality. I can certainly make the change if the consensus is to change it.

My 2 cents is "legacy" makes sense, the regular P2PKH hashes are no longer "standard" in any meaningful sense since they aren't supported by JM any longer. But I personally feel "bech32" may be confusing to less technical users, and "segwit" is pretty straightforward and in line with how it's referenced elsewhere in JM code. But I'll name these whatever is requested.

Reply:
Yes, "standard" is what I don't like most here. @AdamISZ , what do you think?

Reply:
I think we should use the same as what's in `BaseWallet.get_txtype()`, where we match the three TYPE_* vars to a string, since we've used that in a number of other places - although I agree in future it should be done in a smarter way, with a human readable string matched to each scriptpubkey type.

We could turn the existing `TYPE_*` vars into some kind of struct/dict/object so that `BaseWallet.get_txtype()` just returns <object>.human_readable_string() or whatever, and  then the option here could use the same human readable string.

I also agree bech32 is just bad, for the same reason. While `p2wpkh` is a bit hard for some users to read, it is precise, which is good.

Reply:
@AdamISZ , ahh, yes, using the same as in `BaseWallet` is good idea, agree! So it's "p2pkh", "p2sh-p2wpkh", "p2wpkh". These are correct terms and if user has problems with reading / understanding them, he should not be dealing with private keys directly anyway.

Reply:
Yeah. Moreover I want to think a bit more about this PR re: mixed spk types in wallet. It occurs to me this could be problematic w.r.t. coin selection in coinjoins, and also in utxo usage in podle. I always have an intuition to be conservative in this area, which is why I originally wrote "don't let them even try to import other spk types", but I never really thought through what the consequences may be. We all should do that, I think :)

Reply:
@AdamISZ Is there any change I should make to this PR for now?

With regard to consequences, I'm not aware of any. As I understand it, the outputs will be just as mixed as long as they all have the same sizes and key types. But I don't have the deepest understanding of coinjoins, particularly with regards to PODLE.

If indeed podle is the only concern, it should still be fine for makers to import different key types, right? If so, maybe there can be an additional warning not to mix keys types for takers.

Reply:
Sorry for late reply, been a bit overwhelmed last couple of days.

> With regard to consequences, I'm not aware of any. As I understand it, the outputs will be just as mixed as long as they all have the same sizes and key types.

OK expanding it a bit: On this first point, I'm saying, if you allow a user to import a p2sh address into a p2wpkh JM wallet, then it will be (at first), the only p2sh address in there; when a coinjoin happens, the wallet will call `select_utxos` and will (sometimes) pick that imported key to use as part of its set of inputs; so there'll be a non-p2wpkh input in the coinjoin. It's a reasonable question, I think, whether this represents a flag that could be leaking privacy. I'm not sure but I kind of don't like it. In particular, even if I come up with reasoning so that *I* think it's fine for this to happen (and again, I'm not yet really sure), I'm concerned that users will (a) not realize that this could happen and (b) not like it, seeing it as a potential privacy leak.

on PoDLE: so to be super-clear, if a user has used the imported key to generate the PoDLE commitment, it will get rejected:

https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/a5e886944c9ad66dc492ec9ca3caa90aa4f64bb5/jmclient/jmclient/maker.py#L91

... but I think this is a much less important issue, because I think it's reasonable just block PoDLEs from being create-able with imported keys (unless I missed something, should be easy and should solve it OK).

Reply:
Blocking imported keys (or maybe just keys with non-matching key types from the main wallet, if you'd want to allow other imported keys to work) for PoDLEs makes sense.

I see what you're saying about the potential for a privacy leak. I guess my thinking would be that it's a potential privacy leak that users have to opt into by manually importing keys in spite of the warnings/advisories. I can certainly add warnings or expand existing ones as part of this PR if you think that makes sense.